### ISSUE TITLE
MinidInput::connect take self ownership

### ISSUE THREAD

**Issue by MikHulk** [Boddlnagg/midir#90]
Hi,

I don't understand why this [function](https://docs.rs/midir/0.7.0/midir/struct.MidiInput.html#method.connect) takes self ownership. This prevents to reuse MidiInput instance. Moreover I have noticed that MidiInput::new raises an InitError when the same name is reused for the connection. Is there a way to reuse a MidiInput instance ?

**Comment by Boddlnagg** [comment]
I agree that this is probably not the most flexible design, but I think the original reason was that some backend required it, or was otherwise very hard to write without this "limitation". Looking at the code now (a few years after I implemented it), I think it was the ALSA backend, where `MidiInput` is essentially a `snd_seq_t`, and `snd_seq_t` can only handle one open input "channel" at a time, because `snd_seq_t` itself is used as a buffer for the incoming events (s.a. the comment at https://docs.rs/alsa/0.5.0/alsa/seq/struct.Input.html).

If anyone is familiar with ALSA and can chime in and offer a better solution, I'd be happy to change this.

**Comment by MikHulk** [comment]
Yes I have finally found a workaround with message passing. Thank you for the explanation.  I am on MacOSX but I have several linux too. I will try to take a look.

**Comment by sourcebox** [comment]
This seems to be related to #86
